Code/Framework/AzCore/AzCore/Component/ComponentApplication.cpp

```diff
@@ -115,8 +115,9 @@ namespace AZ::Internal
     {
         None, // 0 = no overrides are allowed
         CommandLineOnly, // 1 = registry overrides are allowed from the command line
-        CommandLineAndProject, // 2 = registry overrides are allowed from the command line, engine, gem, and project files
-        CommandLineProjectAndUser // 3 = registry overrides are allowed from the command line, engine, gem, project, and user files
+        ProjectOnly, // 2 = registry overrides are allowed from the engine, gem and project registry files
+        CommandLineAndProject, // 3 = registry overrides are allowed from the command line, engine, gem, and project files
+        CommandLineProjectAndUser // 4 = registry overrides are allowed from the command line, engine, gem, project, and user files
```

Comment:
Minor - this seems like it would be more sustainable as bit flags since each new value is building on top of the previous

Reply:
We could make this a bit flag, but the user interacts with this via adding a CMake Cache Variable when configuring through the CLI or the CMake GUI.

Using a bit flag moves the up the minimum bar of mathematical knowledge the user would need to set these options.

Reply:
I'll update the patch version.
The `ALLOW_SETTINGS_REGISTRY_DEVELOPMENT_OVERRIDES` CMake Cache variable isn't documented on docs.o3de.org, so I doubt there is any user using the option, but I'll update the patch version just in case.